### ISSUE TITLE
GCAM (v7.0) -forcing a target temperature 

### ISSUE THREAD

**Issue by Yangmeansocean** [JGCRI/gcam-core#413]
I am exploring the GCAM model, with its newest version 7.0. However, I think my question is not related to the version but rather applies to the model itself. 

After reading the document (https://jgcri.github.io/gcam-doc/user-guide.html#target-finder): When running the target finder, as is the case when running any policy, users should start from the configuration_policy.xml to ensure **the market structures are properly set up**.  I struggle to understand how I should set up the market structure, **given my target is 1.5 degree**.

When I looked at the configuration_policy.xml, the default shows that many of the input files use the ssp2 version, while I indeed could find ssp1/3/4/5 versions in the input/gcamdata/xml folder. I wonder if these are the files I should use instead, given that SSP2 (the default ones being used) is simply not in line with my 1.5 degree target. 

I ran the model, without addressing the **the market structures are properly set up** issue, but only specified my policy-target in the configuration_policy.xml file: 
policy-target-file: policy_target_1p9_spa1.xml
and 
find-path: 1

and in the configuration.xml file: 
policy-target-file: policy_target_1p9_spa1.xml
and 
find-path: 1

The model seemed to iterate a few times before results were written (as compared to the default run using the configuration_ref), however, tthe results did not changed, as compared to the reference, if I look at co2 emissions or global temperature change over time (for example, by 2100 the temp difference was still 3.6). 

Could anyone help me to solve this issue? Many thanks in advance. @Amsterdam

**Comment by pkyle** [comment]
The file `policy_target_1p9_spa1.xml` is really intended to be bundled with others; it is operating on a market called `CO2_LTG` (where LTG stands for long-term global) which hasn't been otherwise defined. If you look at the file `exe/batch_SSP_SPA1.xml`, that should give you a better template to start from. It will also be using the SSP1 socio-economic and technological background which I assume is what is wanted here.
The way I'd recommend doing this is to start from `configuration_ssp.xml`, assign the `BatchFileName` to `batch_SSP_SPA1.xml`, save it as `configuration.xml`, and then within `batch_SSP_SPA1.xml`, delete all of the rows in the `policy-target-runner` section that correspond to scenarios that you don't want (i.e., keep only the 1.9 scenario).
Note that 1.9 is a hard target; this might not be the end of your difficulties, but at least it should be the correct set-up!

**Comment by Yangmeansocean** [comment]
> The file `policy_target_1p9_spa1.xml` is really intended to be bundled with others; it is operating on a market called `CO2_LTG` (where LTG stands for long-term global) which hasn't been otherwise defined. If you look at the file `exe/batch_SSP_SPA1.xml`, that should give you a better template to start from. It will also be using the SSP1 socio-economic and technological background which I assume is what is wanted here. The way I'd recommend doing this is to start from `configuration_ssp.xml`, assign the `BatchFileName` to `batch_SSP_SPA1.xml`, save it as `configuration.xml`, and then within `batch_SSP_SPA1.xml`, delete all of the rows in the `policy-target-runner` section that correspond to scenarios that you don't want (i.e., keep only the 1.9 scenario). Note that 1.9 is a hard target; this might not be the end of your difficulties, but at least it should be the correct set-up!

Thank you very much! I followed your suggestion and started a batch run last night. So far my laptop (not a bad one) is running for more than 14 hours. what I am missing from the exe environment is the total number of scenario dispatches (now its at #62) to estimate when the model can stop. It is really because of a lack of comprehensive understanding of how the batch run was determined. Before I hit run, I was simply thinking of using the results to understand what scenarios were generated with my single temperature goal. but now it seems like the (time) cost is indeed high and maybe you can direct me to determine given the specified batch run, how many model runs would there be in total? Cheers!

**Comment by pkyle** [comment]
Well, if you're on scenario dispatch number 62, that's about 50 more scenario dispatches than I've ever seen, so I don't think there's anything good happening! There are probably just a lot of solution failures, and each run isn't producing any useful information that helps get closer to the target. I don't know; I haven't ever run a 1.9, and we don't include that scenario in our continuous integration testing. We do include 2.6, so that should work, and perhaps you should just cut off the 1.9 run and try 2.6, and only ratchet up the stringency once you know that the scenario set-up works.

**Comment by Yangmeansocean** [comment]
> Well, if you're on scenario dispatch number 62, that's about 50 more scenario dispatches than I've ever seen, so I don't think there's anything good happening! There are probably just a lot of solution failures, and each run isn't producing any useful information that helps get closer to the target. I don't know; I haven't ever run a 1.9, and we don't include that scenario in our continuous integration testing. We do include 2.6, so that should work, and perhaps you should just cut off the 1.9 run and try 2.6, and only ratchet up the stringency once you know that the scenario set-up works.

Exactly. now after some workout and coming back it is at scenario dispatch #86. I remember last night i did not see any error 'the following model periods did not solve: 21' but today i saw a lot. I guess i should halt the current run and check what you recommended. 

I also have a question: does it mean that, given the newly added/modified relationships/data in v7. 1.5 degree as a target is simply not achievable?

**Comment by pkyle** [comment]
I don't know; again the 1.9 W/m2 target isn't part of our standard testing suite. Still one thing that might make it harder to solve such an extreme scenario is the assumed negative emissions budget specified in `negative_emissions_budget.xml`, which can be commented out or deleted from the `batch_SSP_REF.xml` file. Please post whatever you find, as I don't know that anyone else has tried running this scenario in years!

**Comment by rp9823** [comment]
Hi guys! I'm sorry for joining the discussion but I have a few doubts about topics which are really close to yours. More specifically I'm trying to run the 2.6 and 1.9 and I'd like to check whether I'm doing everything properly. I already run the 4.5 and 3.7 by simply using forcing_4p5 and forcing_3p7 respectively as policy target files in the configuration_policy.xml . The results are satisfying but here comes the first question:
in the configuration_policy.xml appear 4 policies in the scenario components: carbon_tax_0.xml, spa14_tax.xml, 2025_target_finder_phasein.xml, global_uct_phasein_no_constraint.xml, are they necessary in order for the model to run properly the 4p5 and 3.7 scenarios?
furthermore I noticed that among the input/policy files there is the forcing_4p5 but also the policy_target_4p5_spa0,1,3,4,23, the same is true for 3p7, 6p0 and 2p6 while for 1p9 there's only the policy_target "version", could you please tell me what are the ones to be used for the simulation?
Finally, is there a map linking the different SSPs to the RCPs? for example SSP2 for 4p5 and SSP1 for 1p9 

Please let me know if you can help me and thanks a lot in advance!

**Comment by sy123354** [comment]
> 嗨，大家好！很抱歉参加讨论，但我对与您非常接近的话题有一些疑问。更具体地说，我正在尝试运行 2.6 和 1.9，我想检查我是否正确执行所有操作。我已经通过简单地使用 forcing_4p5 和 forcing_3p7 作为configuration_policy.xml中的策略目标文件来运行 4.5 和 3.7。结果令人满意，但第一个问题来了：在configuration_policy.xml场景中，场景组件中出现了 4 个策略：carbon_tax_0.xml、spa14_tax.xml、2025_target_finder_phasein.xml、global_uct_phasein_no_constraint.xml，它们是否是模型正常运行 4p5 和 3.7 场景所必需的？此外，我注意到在输入/策略文件中有forcing_4p5，但也有 policy_target_4p5_spa0、1、3、4、23,3p7、6p0 和 2p6 也是如此，而 1p9 只有policy_target“版本”，您能告诉我用于模拟的是什么吗？最后，是否有一张地图将不同的 SSP 链接到 RCP？例如，SSP2 用于 4p5，SSP1 用于 1p9
> 
> 如果您可以帮助我，请让我知道，并提前非常感谢！

I have encountered the same problem mentioned in the previous comment. Have you found a solution?

**Comment by pkyle** [comment]
I'm not exactly sure what the question is; there seemed to be issues of model implementation as well as theoretical questions about the intersection of the SSP's and the RCP's and which combinations therein are possible. As a description on the theoretical part, the SSP’s define future socioeconomic and technological conditions that are relevant for determining the difficulty of adapting to and mitigating climate change. The RCP's describe the atmospheric conditions over time that are relevant for determining climate change impacts. For purposes of modeling in GCAM, the RCP’s are defined by year 2100 average radiative forcing, which is estimated based on the concentrations of all substances in the atmosphere in the year 2100 that are relevant for the calculation of radiative forcing.
In principle, any SSP could be combined with any RCP; however as a practical matter, many combinations are not possible in the model. Just from a quick Google search, here is a more lengthy explanation that also shows the combinations and their relative feasibility in the models: https://climatescenarios.org/primer/mitigation/
For implementation, I don't recommend using configuration_policy.xml as a starting point; instead I would use configuration_ssp.xml, adjusting the `BatchFileName` (currently `batch_SSP_REF.xml`), to the different SPA files (e.g., SPA1 is SSP1 with global mitigation policy), and then at the bottom of the batch file where the radiative forcing targets are defined, just delete the levels that you don't want.
